src/crypto/blake2s/mod.rs

```diff
@@ -323,15 +355,21 @@ impl Blake2s {
 
         self.hash_block(true);
 
-        let s = unsafe { mem::transmute::<[u32; 8], [u8; 32]>(self.state) };
+        // Convert the internal state from a [u32; 8] to a [u8; 32].
+        let mut s = [0u8; 32];
+        for i in 0..self.outlen {
```

Comment:
Why is it `outlen`, and not a constant like before?

Reply:
The output length is variable. We used to always return a constant amount, but that's bad because we were giving data to higher layers they shouldn't have.

No I'm not doing that. This isn't an optimization target, so I want it simple and readable.